src/libraries/System.Collections/tests/Generic/OrderedDictionary/OrderedDictionary.Generic.Tests.cs

```diff
@@ -187,6 +187,80 @@ public void TryAdd_ItemAlreadyExists_DoesNotInvalidateEnumerator()
 
             Assert.True(valuesEnum.MoveNext());
         }
+
+        [Fact]
+        public void TryAdd_ItemAlreadyExists_IndexCorrect()
```

Comment:
Could you also add a couple of tests validating that we throw when the key is null? There should already be tests for the existing overloads, typically we just group assertions for all overloads in the same test.